### PR TITLE
Lux protected themes

### PR DESCRIPTION
**Pull request by rmichaelis**
Could you please review.

### CODE REVIEW

File: geoportailv3/static/js/catalog/catalogdirective.js [hunk 1]
```diff
@@ -114,9 +114,7 @@ app.CatalogController.prototype.setTree_ = function() {
        * @param {Object} tree Tree object for the theme.
        */
       function(tree) {
-        if (!goog.isNull(tree)) {
-          this['tree'] = tree;
-        }
```
Comment (elemoine) [inline geoportailv3/static/js/catalog/catalogdirective.js]: Are you sure about removing this? I think there was a good reason for this test.

File: geoportailv3/static/js/themeswitcher/themeswitcherdirective.js [hunk 2]
```diff
@@ -37,12 +39,27 @@ app.module.directive('appThemeswitcher', app.themeswitcherDirective);
...
  */
-app.ThemeswitcherController = function(ngeoLocation, appThemes) {
+app.ThemeswitcherController = function(gettextCatalog, ngeoLocation,
+    appThemes, appNotify) {
+
+  /**
+   * @type {function(string, Object<string, *>): string}
+   * @private
+   */
+  this.translate_ = goog.bind(gettextCatalog.getString, gettextCatalog);
```
Comment (elemoine) [inline geoportailv3/static/js/themeswitcher/themeswitcherdirective.js]: Why just store the `gettextCatalog` service on the instance?

File: geoportailv3/static/js/themeswitcher/themeswitcherdirective.js [hunk 3]
```diff
@@ -132,14 +155,26 @@ app.ThemeswitcherController.prototype.setThemes_ = function() {
...
-          this.switchTheme(app.ThemeswitcherController.DEFAULT_THEME_);
+          this.appThemes_.isThemePrivate(this['currentTheme']).then(goog.bind(
+              /**
+               * @param {angular.$http.Response} resp Ajax response.
+               */
+              function(resp) {
+                if (resp.data['is_private'] === true) {
+                  this.appNotify_(this.translate_(this.privateThemeMsg_, {}));
+                  this['userOpen'] = true;
+                }else {
```
Comment (elemoine) [inline geoportailv3/static/js/themeswitcher/themeswitcherdirective.js]: Space missing after `}`.

File: geoportailv3/static/js/catalog/catalogdirective.js (hunk 1 above)
Comment (rmichaelis) [inline geoportailv3/static/js/catalog/catalogdirective.js]: It seems to work.
What will be a better way to notify the layer manager, that the layer tree does not exist anymore ?

Comment (elemoine) [inline geoportailv3/static/js/catalog/catalogdirective.js]: Ok.

I just remembered that there was a good reason for this test, but it's not clear to me anymore.

**Comment by elemoine** [comment]
LGTM.

**Comment by elemoine** [comment]
You could clean up the history and squash your commits into one.

**Pull request merged**